### DISCUSSION TITLE
pay invoice with outgoing channel

### DISCUSSION THREAD

**Discussion by frbitten** [Ride-The-Lightning/RTL#1233] [C-Lightning]
Is it possible to pay an invoice and choose the funds exit channel? in RTL for LND there is this functionality but when switching to CLN I was surprised and did not find this functionality.

Is there any way to enable this functionality in CLN?

**Comment by ShahanaFarooqui** [comment]
Core Lightning does not provide this feature in their API, so it is not possible to implement it in the UI.

**Reply by frbitten** [reply]
> Core Lightning does not provide this feature in their API, so it is not possible to implement it in the UI.

Core Lightning does implement this functionality. Just a matter of the UI enabling functionality.
You can see in this link how the pay command works: [https://lightning.readthedocs.io/lightning-pay.7.html](https://lightning.readthedocs.io/lightning-pay.7.html)

In other words, it only allows me to select the channel I want to output and include all others in the exclude.
